src/tool_writeout_json.c

```diff
@@ -0,0 +1,178 @@
+  {"response_code", JSON_LONG, CURLINFO_RESPONSE_CODE},
+  {"http_connect", JSON_LONG, CURLINFO_HTTP_CONNECTCODE},
+  {"time_total", JSON_DOUBLE, CURLINFO_TOTAL_TIME},
+  {"time_namelookup", JSON_DOUBLE, CURLINFO_NAMELOOKUP_TIME},
+  {"time_connect", JSON_DOUBLE, CURLINFO_CONNECT_TIME},
+  {"time_appconnect", JSON_DOUBLE, CURLINFO_APPCONNECT_TIME},
+  {"time_pretransfer", JSON_DOUBLE, CURLINFO_PRETRANSFER_TIME},
+  {"time_starttransfer", JSON_DOUBLE, CURLINFO_STARTTRANSFER_TIME},
+  {"size_header", JSON_LONG, CURLINFO_HEADER_SIZE},
+  {"size_request", JSON_LONG, CURLINFO_REQUEST_SIZE},
+  {"size_download", JSON_DOUBLE, CURLINFO_SIZE_DOWNLOAD},
+  {"size_upload", JSON_DOUBLE, CURLINFO_SIZE_UPLOAD},
+  {"speed_download", JSON_DOUBLE, CURLINFO_SPEED_DOWNLOAD},
+  {"speed_upload", JSON_DOUBLE, CURLINFO_SPEED_UPLOAD},
+  {"content_type", JSON_STRING, CURLINFO_CONTENT_TYPE},
+  {"num_connects", JSON_LONG, CURLINFO_NUM_CONNECTS},
+  {"time_redirect", JSON_DOUBLE, CURLINFO_REDIRECT_TIME},
+  {"num_redirects", JSON_LONG, CURLINFO_REDIRECT_COUNT},
+  {"ftp_entry_path", JSON_STRING, CURLINFO_FTP_ENTRY_PATH},
+  {"redirect_url", JSON_STRING, CURLINFO_REDIRECT_URL},
+  {"ssl_verify_result", JSON_LONG, CURLINFO_SSL_VERIFYRESULT},
+  {"proxy_ssl_verify_result", JSON_LONG, CURLINFO_PROXY_SSL_VERIFYRESULT},
+  {"filename_effective", JSON_FILENAME, CURLINFO_NONE},
+  {"remote_ip", JSON_STRING, CURLINFO_PRIMARY_IP},
+  {"remote_port", JSON_LONG, CURLINFO_PRIMARY_PORT},
+  {"local_ip", JSON_STRING, CURLINFO_LOCAL_IP},
+  {"local_port", JSON_LONG, CURLINFO_LOCAL_PORT},
+  {"http_version", JSON_VERSION, CURLINFO_HTTP_VERSION},
+  {"scheme", JSON_STRING, CURLINFO_SCHEME},
+  {NULL, JSON_NONE, CURLINFO_NONE}
+};
+
+static const char* http_version[] = {
+  "0",   /* CURL_HTTP_VERSION_NONE */
+  "1",   /* CURL_HTTP_VERSION_1_0 */
+  "1.1", /* CURL_HTTP_VERSION_1_1 */
+  "2"    /* CURL_HTTP_VERSION_2 */
+  "3"    /* CURL_HTTP_VERSION_3 */
+};
+
+static void json_escape(FILE* stream, const char* in)
+{
+  const char* i = in;
+  const char* in_end = in + strlen(in);
+
+  for (; i < in_end; i++) {
+    switch (*i) {
+    case '\\':
+      fputs("\\\\", stream);
+      break;
+    case '\"':
+      fputs("\\\"", stream);
+      break;
+    case '\b':
+      fputs("\\b", stream);
+      break;
+    case '\f':
+      fputs("\\f", stream);
+      break;
+    case '\n':
+      fputs("\\n", stream);
+      break;
+    case '\r':
+      fputs("\\r", stream);
+      break;
+    case '\t':
+      fputs("\\t", stream);
+      break;
+    default:
+      if (*i < 32) {
+        fprintf(stream, "u%04x", *i);
+      } else {
+        fputc(*i, stream);
+      }
+      break;
+    };
+  }
+}
+
+
+void ourWriteOutJSON(CURL *curl, struct OutStruct *outs, FILE *stream)
+{
+  char *stringp = NULL;
+  long longinfo;
+  double doubleinfo;
+  int i;
+
+  fputs("{", stream);
+  for (i = 0; keys[i].name != NULL; i++) {
+    const char* k = keys[i].name;
```

Comment:
Couldn't you just rename this to key? Makes more sense as json pretty much looks like a map and these are common terms for it